src/main/kotlin/org/partiql/scribe/targets/spark/SparkAstToSql.kt

```diff
@@ -1,6 +1,167 @@
+                    )
+            }
+        h = visitExprWrapped(node.expr, h)
+        h = if (node.asAlias != null) h concat " AS ${node.asAlias!!.sql()}" else h
+        // AT and BY should be stopped by feature validation.
+        return h
+    }
+
+    override fun visitIdentifier(
+        node: Identifier,
+        tail: SqlBlock,
+    ): SqlBlock {
+        val path =
+            when (node.hasQualifier()) {
+                true -> {
+                    val qualifier = node.qualifier.fold("") { acc, part -> acc + "${part.sql()}." }
+                    qualifier + node.identifier.sql()
+                }
+                false -> node.identifier.sql()
+            }
+        return tail concat path
+    }
+
+    override fun visitSelectItemExpr(
+        node: SelectItem.Expr,
+        tail: SqlBlock,
+    ): SqlBlock {
+        var h = tail
+        h = visitExprWrapped(node.expr, h)
+        h = if (node.asAlias != null) h concat " AS ${node.asAlias!!.sql()}" else h
+        return h
+    }
+
+    override fun visitPathStepElement(
+        node: PathStep.Element,
+        tail: SqlBlock,
+    ): SqlBlock {
+        val key = node.element
+        return if (key is ExprLit && key.lit.code() == Literal.STRING) {
+            val elemString = key.lit.stringValue()
+            val stepField = exprPathStepField(Identifier.Simple.regular(elemString))
+            visitPathStepField(stepField, tail)
+        } else {
+            super.visitPathStepElement(node, tail)
+        }
+    }
+
+    override fun visitPathStepField(
+        node: PathStep.Field,
+        tail: SqlBlock,
+    ): SqlBlock {
+        return tail concat ".${node.field.sql()}"
+    }
+
+    /**
+     * Type mappings for Spark
+     * INT2 -> SMALLINT
+     * INT4 -> INT
+     * INT8 -> BIGINT
+     * DOUBLE_PRECISION -> DOUBLE
+     */
+    override fun visitDataType(
+        node: DataType,
+        tail: SqlBlock,
+    ): SqlBlock {
+        return when (node.code()) {
+            DataType.INT2 -> tail concat "SMALLINT"
+            DataType.INT4 -> tail concat "INT"
+            DataType.INT8 -> tail concat "BIGINT"
+            DataType.DOUBLE_PRECISION -> tail concat "DOUBLE"
+            else -> super.visitDataType(node, tail)
+        }
+    }
+
+    override fun visitExprBag(
+        node: ExprBag,
+        tail: SqlBlock,
+    ): SqlBlock {
+        return tail concat list(this, "(", ")") { node.values }
+    }
+
+    override fun visitLiteral(
+        node: Literal,
+        ctx: SqlBlock,
+    ): SqlBlock {
+        if (node.code() == Literal.TYPED_STRING) {
+            if (node.dataType().code() == DataType.TIME || node.dataType().code() == DataType.TIME_WITH_TIME_ZONE) {
+                listener.reportAndThrow(
+                    ScribeProblem.simpleError(
+                        ScribeProblem.UNSUPPORTED_AST_TO_TEXT_CONVERSION,
+                        "Attempting to print ${node.stringValue()}. Time type is not supported in Spark.",
```

Comment:
`Attempting to print` feels like it talks to much about implementation details (i.e., `print`), not about the error here.

Maybe `Time type is not supported in Spark. Trying to convert literal ${node.stringValue()}`?